x-pack/plugins/lens/public/indexpattern_datasource/utils.tsx

```diff
@@ -114,58 +178,52 @@ export function getPrecisionErrorWarningMessages(
         const currentColumn = currentLayer?.columns[column.id];
         if (currentLayer && currentColumn && checkColumnForPrecisionError(column)) {
           const indexPattern = state.indexPatterns[currentLayer.indexPatternId];
+          const currentColumnIsTerms = isColumnOfType<TermsIndexPatternColumn>(
+            'terms',
+            currentColumn
+          );
           const isAscendingCountSorting =
-            isColumnOfType<TermsIndexPatternColumn>('terms', currentColumn) &&
+            currentColumnIsTerms &&
             currentColumn.params.orderBy.type === 'column' &&
             currentColumn.params.orderDirection === 'asc' &&
             isColumnOfType<CountIndexPatternColumn>(
               'count',
               currentLayer.columns[currentColumn.params.orderBy.columnId]
             );
           const usesFloatingPointField =
-            isColumnOfType<TermsIndexPatternColumn>('terms', currentColumn) &&
+            currentColumnIsTerms &&
             !supportsRarityRanking(indexPattern.getFieldByName(currentColumn.sourceField));
           const usesMultipleFields =
-            isColumnOfType<TermsIndexPatternColumn>('terms', currentColumn) &&
-            (currentColumn.params.secondaryFields || []).length > 0;
-          if (!isAscendingCountSorting || usesFloatingPointField || usesMultipleFields) {
+            currentColumnIsTerms && (currentColumn.params.secondaryFields || []).length > 0;
+
+          if (
+            currentColumnIsTerms &&
```

Comment:
@flash1293 I added this check which causes a test to fail. Just to confirm, do precision errors apply to aggregation types other than `terms`?

Reply:
You are right, the test needs to be updated - adding a new terms column and simulating the precision warning on that one should be the best fix.

Reply:
So really, `checkColumnForPrecisionError` will never return true on a non-terms agg?

Reply:
It will never return true on a non-terms "operation", I think on the agg level it can happen at least for terms and multi_terms.